pages/reference/OS/secure-boot-and-full-disk-encryption/overview.md

```diff
@@ -0,0 +1,121 @@
+- **Non-flasher images**: Used when the devices did not expose the internal storage. The flasher image boots from an external storage medium and then the os is programmed to the required internal storage 
+
+For secure boot-enabled devices, the distinction blurs - all secure boot-enabled devices have flasher images. Moreover, given that the image not only writes the os to disk but also does other tasks such as enrolling required can be considered as `installer` images. So, secure boot enabled devices provide an `installer` image instead that can be programmed either on external or internal storage, in the latter case running from memory and performing the installation on the same disk it is booted from.
+
+The secure boot installer is itself using signed binaries and can be used to re-program locked devices. Re-running the installer will irrevocably lose all data on the disks as the encryption key is erased.
+
+
+__Note:__ This imposes a limitation on the specifications for device types to support secure boot - they must have enough memory to hold the raw image that is being programmed into disk. BalenaOS reduces this restriction by using compressed memory to hold the raw image, but still, memory constrained devices will not be able to support secure boot with the current design.
+
+
+### Partition Layout and Disk Encryption
+Balena uses the `dm-crypt` Linux kernel subsystem and `device-mapper` framework to encrypt block devices. Both direct `dm-crypt` and `LUKS` (Linux Unified Key Setup) are supported.
+
+A secure boot installation also alters the default balenaOS partition table as the boot partition is split into two:
+- An unencrypted boot partition which contains the signed balena bootloader binary and other essential artifacts needed to run it.
+- An encrypted boot partition that mounts in the traditional `/mnt/boot` path and contains all other artifacts, including confidential files like `config.json` which contains API keys.
+
+
+## Boot Chain of Trust
+
+The notion of a chain of trust arises because each stage in the boot process validates the integrity of the subsequent stage before executing it. If at any point a link in this chain is corrupted, unsigned, or otherwise fails validation, the boot process halts or reverts to a known-good state. This sequential validation ensures that if the first link is trusted, every subsequent link remains trustworthy, preventing unauthorized or tampered code from ever running.
+
+The root of trust is always the device’s `bootROM`. `bootROM` refers to the very first piece of firmware that runs when a device powers on. It resides in read-only memory (ROM) that is physically embedded in the processor or system-on-chip (SoC). Because the code stored in bootROM is difficult or impossible to modify (without physically altering the chip), it serves as the hardware-enforced starting point or “root of trust” for the boot process. 
```

Comment:
This is critical to whole security and requires proper checks of the chosen hardware platform, apply bios updates, and turn on bios security features.

There's a requirement of having a hardware secure element (TPM, or equivalent depending on the platform). And that a fTPM is considered more secure than a discrete TPM to prevent bus sniffing.

Reply:
@aethernet anything to change/add/remove in this section?

i think the "secure element" varies by platform and this document right now aims to be generic. we will have device-type dependent docs where perhaps we can highlight the specific secure elements and their limitations.

or perhaps you believe it is better we mention embedded hardware secure element vs external secure element?

Reply:
> This is critical to whole security and requires proper checks of the chosen hardware platform, apply bios updates, and turn on bios security features.
> 
> There's a requirement of having a hardware secure element (TPM, or equivalent depending on the platform). And that a fTPM is considered more secure than a discrete TPM to prevent bus sniffing.

The TPM is not used to ensure the chain of trust, so I don't think this is the correct place for this piece of information.

Also "apply bios updates" is contradictory to "It resides in read-only memory (ROM) ... the code stored in bootROM is difficult or impossible to modify (without physically altering the chip)".